Mono.Addins/Mono.Addins.Database/DatabaseConfiguration.cs

```diff
@@ -121,6 +117,34 @@ public AddinStatus[] GetPendingUninstalls ()
 		
 		public static DatabaseConfiguration Read (string file)
 		{
+			var config = ReadInternal (file);
+			// Try to read application level config to support disabling add-ins by default.
+			var appConfig = ReadAppConfig ();
+
+			if (appConfig == null)
+				return config;
+
+			// Overwrite app config values with user config values
+			foreach (var entry in config.addinStatus)
+				appConfig.addinStatus [entry.Key] = entry.Value;
+
+			return appConfig;
+		}
+		
+		public static DatabaseConfiguration ReadAppConfig()
+		{
+			var exePath = System.Reflection.Assembly.GetExecutingAssembly ().Location;
```

Comment:
You are right, it returns the current assembly. That's what you get when you copy-paste code :smile: I'll change the name.